gh-codeql

```diff
@@ -156,10 +166,41 @@ function set_version() {
             error "Unknown version: '$1'."
         fi
     fi
+    echo "$version"
+}
+
+function set_version() {
+    local version=$(validate_version "$1")
     download "$version"
     gh config set extensions.codeql.version "$version" 2> /dev/null # Ignore a warning about unrecognized custom keys
 }
 
+function set_local_version() {
+    local version=$(validate_version "$1")
+    download "$version"
+    echo "$version" > .codeql-version
+}
+
+function install_stub() {
+    local bindir="$1"
+    if [ -z "$bindir" ]; then
+        bindir="/usr/local/bin"
+    fi
+
+    if [ ! -e "$bindir" ]; then
+        error "The directory $bindir doesn't exist, cannot install stub!"
```

Comment:
Shall we suggest passing an argument to `gh codeql install-stub`? A user who has just run that may not be aware that is a possibility.

Reply:
Good suggestion. I've adjusted the error message and added an extra write permission check.